packages/bpk-component-snippet/src/BpkSnippet.tsx

```diff
@@ -0,0 +1,121 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { MouseEvent } from 'react';
+
+import { BpkButtonV2, BUTTON_TYPES } from '../../bpk-component-button';
+import BpkText, {
+  TEXT_STYLES,
+  type TextStyle,
+} from '../../bpk-component-text/src/BpkText';
+import { cssModules } from '../../bpk-react-utils';
+
+import type { ButtonType } from '../../bpk-component-button/src/BpkButtonV2/common-types';
+
+import STYLES from './BpkSnippet.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+export type Props = {
+  src: string;
+  altText: string;
+  headline?: string | null;
+  subheading?: string | null;
+  bodyText?: string | null;
+  buttonText?: string | null;
+  bodyStyle?: typeof TEXT_STYLES.bodyDefault | typeof TEXT_STYLES.bodyLongform;
+  buttonStyle?: ButtonType;
+  headlineStyle?: TextStyle;
+  desktopLayout?: 'imageLeft' | 'imageRight' | 'vertical';
```

Comment:
I would change this to be an enum defined above this as

```tsx

export const DESKTOP_LAYOUTS = {
   imageLeft: 'imageLeft',
   imageRight: 'imageRight',
   vertical: 'vertical'
} as const;
```

Then this can be 
```suggestion
  desktopLayout?: (typeof DESKTOP_LAYOUTS)[keyof typeof DESKTOP_LAYOUTS];
```

Reply:
Updated!